webapp/src/store/Store.ts

```diff
@@ -1,7 +1,59 @@
+import fs from 'fs/promises';
+
 import { ProjectStore } from '@/store/ProjectStore';
+import MessageAdapterFactory from '@/utils/adapters/MessageAdapterFactory';
+import YamlTranslationAdapter from '@/utils/adapters/YamlTranslationAdapter';
+import { LyraProjectConfig } from '@/utils/lyraConfig';
+import { StoreData } from './types';
+
+const FILE_PATH = './store.json';
 
 export class Store {
   private data = new Map<string, ProjectStore>();
+  private initialState: Record<string, StoreData | undefined> = {};
+
+  public static async getProjectStore(
+    lyraProjectConfig: LyraProjectConfig,
+  ): Promise<ProjectStore> {
+    if (!globalThis.store) {
+      globalThis.store = new Store();
+      await globalThis.store.loadFromDisk();
+    }
+
+    if (!globalThis.store.hasProjectStore(lyraProjectConfig.absPath)) {
+      const initialProjectState =
+        globalThis.store.initialState[lyraProjectConfig.absPath];
+
+      const projectStore = new ProjectStore(
+        MessageAdapterFactory.createAdapter(lyraProjectConfig),
+        new YamlTranslationAdapter(lyraProjectConfig.absTranslationsPath),
+        initialProjectState,
+      );
+      globalThis.store.addProjectStore(lyraProjectConfig.absPath, projectStore);
+    }
+
+    return globalThis.store.getProjectStore(lyraProjectConfig.absPath);
+  }
+
+  public static async persistToDisk(): Promise<void> {
+    if (!globalThis.store) {
+      return;
+    }
+
+    const payload = globalThis.store.toJSON();
+    const json = JSON.stringify(payload);
+
+    await fs.writeFile(FILE_PATH, json);
```

Comment:
It is unsafe to use `writeFile` multiple times on the same file without waiting for the promise to be settled.

The probability of concurrent `updateTranslation` is high so we will have to find a resolution for this before merging.